pkgs/shells/fish/default.nix

```diff
@@ -1,24 +1,31 @@
 {
   stdenv,
   lib,
-  fetchurl,
-  fetchpatch,
+  fetchFromGitHub,
   coreutils,
-  which,
+  darwin,
+  glibcLocales,
   gnused,
   gnugrep,
   groff,
   gawk,
   man-db,
+  ninja,
   getent,
   libiconv,
   pcre2,
+  pkg-config,
+  sphinx,
   gettext,
   ncurses,
```

Comment:
I believe ncurses is no longer a compile-time dependency, see [the current build instructions](https://github.com/fish-shell/fish-shell/?tab=readme-ov-file#dependencies-1) (vs [those of 3.7.1](https://github.com/fish-shell/fish-shell/blob/3.7.1/README.rst#dependencies-1)).

Reply:
It's referenced on L318 and is (only) used in `preCheck`; from the current build instructions:

> Running fish requires:
>
> *    A terminfo database, typically from curses or ncurses (preinstalled on most *nix systems) - this needs to be the directory tree format, not the "hashed" database. If this is unavailable, fish uses an included xterm-256color definition.

Reply:
I think it is still required for terminfo stuff so it better to keep it their just in case it causes terminfo issues.

Also on [readme section](https://github.com/fish-shell/fish-shell/?tab=readme-ov-file#dependencies) for more info why it's should be required. But I think it should be verified if it's actually needed.

Reply:
Good catch, thanks!